src/templates/security-scheme-template.js

```diff
@@ -36,99 +41,168 @@ function onClearAllApiKeys() {
     }
-    */
+  }
 
-    // After handeling, remove this listener
-    window.removeEventListener('message', handleMessageEventFn, true);
-    w.close();
-    if (!ev.data) {
-      console.error('RapiDoc: Received no data with authorization message');
-    }
-    if (ev.data.state !== state) {
-      console.warn('RapiDoc: State value did not match.');
-    }
-    if (ev.data.error) {
-      console.warn('RapiDoc: Error while receving data');
-    }
-    if (ev.data) {
-      if (ev.data.responseType === 'code') {
-        console.log(`RapiDoc: AUTH CODE RECEIVED - ${ev.data.code}`);
-        // return res(ev.data.code);
-        const formData = new FormData();
-        formData.append('grant_type', 'authorization_code');
-        formData.append('code', ev.data.code);
-        formData.append('client_id', clientId);
-        formData.append('client_secret', clientSecret);
-        formData.append('redirect_uri', receiveUrlObj.toString());
-        try {
-          const resp = await fetch(tokenUrl, { method: 'POST', body: formData });
-          console.log(`OAUth Token Response Status: ${resp.statusText}:${resp.status}`);
-          const respObj = await resp.json();
-          console.log('Access Token Response: %o', respObj);
-          if (respObj.access_token) {
-            securityObj.finalKeyValue = `${respObj.token_type} ${respObj.access_token}`;
-            this.requestUpdate();
-          }
-        } catch (err) {
-          console.error('RapiDoc: Unable to get access token');
-        }
-      } else if (ev.data.responseType === 'token') {
-        securityObj.finalKeyValue = `${ev.data.token_type} ${ev.data.access_token}`;
-        this.requestUpdate();
-      }
-    }
+  const authorizationUrl = authConfigs.authorizationUrl;
+  const sanitizedAuthorizationUrl = sanitizeUrl(authorizationUrl);
+  const url = [sanitizedAuthorizationUrl, query.join('&')].join(authorizationUrl.indexOf('?') === -1 ? '?' : '&');
+
+  // pass action authorizeOauth2 and authentication data through window
+  // to authorize with oauth2
+
+  let callback;
+  if (flow === 'implicit') {
+    callback = (payload) => preAuthorizeImplicit(payload).then((token) => {
+      // @todo not built yet
+      console.log('Missing functionality to handle token', token);
+    });
+  } else if (authConfigs.useBasicAuthenticationWithAccessCodeGrant) {
+    callback = (payload) => authorizeAccessCodeWithBasicAuthentication(payload).then((token) => {
+      // @todo not built yet
+      console.log('Missing functionality to handle token', token);
+    });
+  } else {
+    callback = (payload) => authorizeAccessCodeWithFormParams(payload).then((token) => {
+      if (!token) return;
+      securityObj.finalKeyValue = `Bearer ${token.access_token}`;
+      this.requestUpdate();
+    });
+  }
+
+  window.Oauth2UIRedirect = {
+    flow,
+    state,
+    redirectUrl,
+    callback,
+    authConfigs,
+    error: (err) => console.error(err),
```

Comment:
`Oauth2UIRedirect` a custom attribute is created for window-to-window communication, Instead I would prefer the standard approach [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) and [`messageEvent`](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent) which currently being used